tasks/gather_ports.yml

```diff
@@ -0,0 +1,165 @@
+---
+# Initialize variables
+- name: Initialize logging_tls_tcp_ports
+  set_fact:
+    logging_tls_tcp_ports: []
+
+- name: Initialize logging_tcp_ports
+  set_fact:
+    logging_tcp_ports: []
+
+- name: Initialize logging_tls_udp_ports
+  set_fact:
+    logging_tls_udp_ports: []
+
+- name: Initialize logging_udp_ports
+  set_fact:
+    logging_udp_ports: []
+
+# Gather ports configured as logging role parameters
+- name: Parameter 'port' values
+  set_fact:
+    logging_tls_tcp_ports: "{{ (logging_inputs + logging_outputs) | d([]) |
+                               selectattr('port', 'defined') |
+                               map(attribute='port') }}"
+
+- block:
+    - name: Parameter 'tcp_port' values (without tls)
+      set_fact:
+        logging_tcp_ports: "{{ logging_tcp_ports | union(__tcp_ports[0]) }}"
+
+    - name: Parameter 'tcp_port' values (with tls)
+      set_fact:
+        logging_tls_tcp_ports: "{{ logging_tls_tcp_ports |
+                                   union(__tcp_ports[1]) }}"
+
+    - name: Parameter 'udp_port' values (without tls)
+      set_fact:
+        logging_udp_ports: "{{ logging_udp_ports | union(__udp_ports[0]) }}"
+
+    - name: Parameter 'udp_port' values (with tls)
+      set_fact:
+        logging_tls_udp_ports: "{{ logging_tls_udp_ports |
+                                   union(__udp_ports[1]) }}"
+
+    - name: Parameter 'server_port' values (without tls)
+      set_fact:
+        logging_tcp_ports: "{{ logging_tcp_ports | union(__server_ports[0]) }}"
+
+    - name: Parameter 'server_port' values (with tls)
+      set_fact:
+        logging_tls_tcp_ports: "{{ logging_tls_tcp_ports |
+                                   union(__server_ports[1]) }}"
+  vars:
+    __tcp_outputs: "{{ logging_outputs | d([]) |
+                    selectattr('tcp_port', 'defined') }}"
+    __tcp_ports: |
+      {% set tcp_ports = [] %}
+      {% set tls_tcp_ports = [] %}
+      {% for output in __tcp_outputs %}
+      {%   if output.tls | d() %}
+      {%     if output.tls -%}
+      {%       set _ = tls_tcp_ports.append(output.tcp_port) %}
+      {%     else -%}
+      {%       set _ = tcp_ports.append(output.tcp_port) %}
+      {%-    endif %}
+      {%   else -%}
+      {%       set _ = tcp_ports.append(output.tcp_port) %}
+      {%-  endif %}
+      {% endfor %}
+      {% set both = [tcp_ports, tls_tcp_ports] %}
+      {{ both }}
```

Comment:
Sadly, this does not work for ansible-2.9. :(

Reply:
what is the issue?

Reply:
Ansible 2.9 requires strict list filter to convert string to list... (ansible-core-2.13 quietly interprets it for me, and I did not notice it... :p)